core/tests/test_state_transition_model.py

```diff
@@ -20,5 +27,149 @@ def test_create_incident_report_creates_new_state_transition(self):
             police_abstract_obtained="Yes",
             submitted_by=self.user,
         )
-        self.assertEqual(AssetIncidentReport.objects.count(), incident_report_count + 1)
-        self.assertEqual(StateTransition.objects.count(), state_transition_count + 1)
+        self.assertEqual(
+            apps.get_model("core", "AssetIncidentReport").objects.count(),
+            incident_report_count + 1,
+        )
+        self.assertEqual(
+            apps.get_model("core", "StateTransition").objects.count(),
+            state_transition_count + 1,
+        )
+
+
+class TestTransitionStateUpdateFromAssetStatusModification(CoreBaseTestCase):
+    """
+    Test updating of transition state when Asset status is modified
+    Update the transition state to CLOSED when an Asset status is changed from (DAMAGED,LOST) to
+    either (AVAILABLE,ALLOCATED)
+    """
+
+    def setUp(self):
+        self.asset = apps.get_model("core", "Asset").objects.create(
+            asset_code="IC0019009",
+            serial_number="SN001000098",
+            model_number=self.test_assetmodel,
+            purchase_date="2019-07-10",
+        )
+        self.incident_report = apps.get_model(
+            "core", "AssetIncidentReport"
+        ).objects.create(asset=self.asset)
+        self.transition_state = apps.get_model(
+            "core", "StateTransition"
+        ).objects.get_or_create(asset_incident_report_id=self.incident_report.id)
+        self.transition_state = self.transition_state[0]
+
+    def test_update_asset_status_from_lost_to_available(self):
+        status = (
+            apps.get_model("core", "AssetStatus")
+            .objects.filter(asset=self.asset)
+            .latest('created_at')
+        )
+        # update status to lost
+        status.current_status = constants.LOST
+        status.save()
+        # update status to available
+        status = (
+            apps.get_model("core", "AssetStatus")
+            .objects.filter(asset=self.asset)
+            .latest('created_at')
+        )
+        status.current_status = constants.AVAILABLE
+        status.save()
+        # verify that transition state has been updated to closed
+        transition_state = apps.get_model("core", "StateTransition").objects.get(
+            asset_incident_report=self.incident_report
+        )
+        self.assertEqual(transition_state.incident_report_state, constants.CLOSED)
+
+    def test_update_asset_status_from_damaged_to_available(self):
+        status = (
+            apps.get_model("core", "AssetStatus")
+            .objects.filter(asset=self.asset)
+            .latest('created_at')
+        )
+
+        # update status to lost
+        status.current_status = constants.DAMAGED
```

Comment:
lol. Comment above says update status to lost. lol

Reply:
Noted and fixed